packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -252,11 +252,13 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
   /// see the discussion there for why that might still be a bad idea.)
   void correctPixels(double value) {
     _pixels = value;
+    _didChangeViewportDimensionOrPixels = true;
```

Comment:
The documentation says "In particular, this is typically called as a response to applyViewportDimension or applyContentDimensions", but here you're writing code that assumes it's called before applyContentDimensions, which seems dubious.

Reply:
Ya, I'm a bit concerned touching something this central since there's a lot of implicit coupling, though all the tests passed. 

I wasn't too sure what's the best way either. The correct behaviour seems to work with long lists because `applyNewDimensions` is called on line 426 to re-kick the start ballistic activity. Perhaps you have a better solution. 

This field seemed like the least invasive as far as I can tell since the current code which uses the field instead of the setter seems to want to defer signaling after this loop here https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/viewport.dart#L1053. I'm assuming the documentation saying it's used in response to applyContentDimensions likely meant applyContentDimensions resets this value to false after consuming it.

Reply:
PTAL, did what we talked about offline and added some docs and an assert.